Basic/APIDiorama/Assets/UseCaseSamples/ProximityChecks/Scripts/ColorManager.cs

```diff
@@ -0,0 +1,70 @@
+using UnityEngine;
+
+namespace Unity.Netcode.Samples.APIDiorama
+{
+    /// <summary>
+    /// Manages the color of a Networked object
+    /// </summary>
+    public class ColorManager : NetworkBehaviour
+    {
+
+        NetworkVariable<Color32> m_NetworkedColor = new NetworkVariable<Color32>();
+        Material m_Material;
+        ProximityChecker m_ProximityChecker;
+
+        void Awake()
+        {
+            m_Material = GetComponent<Renderer>().material;
+            m_ProximityChecker = GetComponent<ProximityChecker>();
+        }
+
+        public override void OnNetworkSpawn()
+        {
+            base.OnNetworkSpawn();
+            if (IsClient)
+            {
+                OnClientColorChanged(m_Material.color, m_NetworkedColor.Value);
+                m_NetworkedColor.OnValueChanged += OnClientColorChanged;
+            }
+        }
+
+        public override void OnNetworkDespawn()
+        {
+            base.OnNetworkDespawn();
+            if (IsClient)
+            {
+                m_NetworkedColor.OnValueChanged -= OnClientColorChanged;
+            }
+        }
+
+        void Update()
+        {
+            if (!IsClient
```

Comment:
I agree we should cut long lines, but that's pushing it :P
Rule of thumb is usually "if it fits on a 1024px large monitor, it'll be fine"

Reply:
Changed! I usually split multiple checks in this way as it makes them more readeable, but if we have a standard for putting everything in one line I'll follow it.